templates/web/base/admin/category-checkboxes.html

```diff
@@ -18,8 +10,31 @@
         <label class="inline" title="[% contact.email | html %]">
           <input type="checkbox" name="contacts[[% contact.id %]]" [% 'checked' IF contact.active %]/>
           [% contact.category %]
+          [% IF contact.group.size > 1 %]<small>([% contact.group.join('; ') | html %])</small>[% END %]
         </label>
       </li>
     [% END %]
   </ul>
+[% END %]
+<fieldset>
+  <legend>
+  [% IF hint %]
+    <div class="admin-hint">
+      <p>[% hint %]</p>
+    </div>
+  [% END %]
+    [% loc('Categories:') %]
+  </legend>
+
+  [% IF category_groups %]
+    [% FOR group IN category_groups %]
+      <h3>[% group.name | html %]</h3>
```

Comment:
If there are categories without a group, then these will be listed under a blank heading here; perhaps default to "No Group" in that case?